src/protocol.rs

```diff
@@ -121,145 +131,70 @@ impl Participant {
-                            &keyshare_private,
-                        )?;
-                        Output::KeyGen(keyshare_publics, keyshare_private)
-                    }
-                    None => Output::None,
-                };
-                Ok((message.id(), public_output, messages))
+                Ok((message.id(), output, messages))
             }
-            MessageType::Presign(_) => {
+            (MessageType::Presign(_), ProtocolType::Presign) => {
                 // Send presign message and existing storage containing auxinfo and
                 // keyshare values that presign needs to operate
-                let outcome =
-                    self.presign_participant
-                        .process_message(rng, message, &self.main_storage)?;
+                let outcome = self
+                    .participant
+                    .process_message(rng, message, &self.input)?;
 
                 let (output, messages) = outcome.into_parts();
-
-                let public_output = match output {
-                    Some(record) => {
-                        self.main_storage.store(
-                            PersistentStorageType::PresignRecord,
-                            message.id(),
-                            self.id,
-                            &record,
-                        )?;
-                        Output::Presign(record)
-                    }
-                    None => Output::None,
-                };
-                Ok((message.id(), public_output, messages))
+                Ok((message.id(), output, messages))
+            }
+            _ => {
+                error!(
+                    "Message type did not match type of this participant: got {:?}, expected {:?}",
+                    message.message_type(),
+                    P::protocol_type()
+                );
+                Err(InternalError::MisroutedMessage)
             }
-            _ => Err(InternalError::MisroutedMessage),
         }
     }
 
-    /// Produces a message to signal to this participant that auxinfo generation
-    /// is ready for the specified identifier.
-    #[instrument(skip_all)]
-    pub fn initialize_auxinfo_message(&self, auxinfo_identifier: Identifier) -> Message {
-        info!("Auxinfo generation is ready.");
-        Message::new(
-            MessageType::Auxinfo(AuxinfoMessageType::Ready),
-            auxinfo_identifier,
-            self.id,
-            self.id,
-            &[],
-        )
-    }
-
-    /// Produces a message to signal to this participant that keyshare
-    /// generation is ready for the specified identifier
+    /// Produce a message to signal to this participant that the protocol can
+    /// begin.
+    ///
+    /// Possible bug: previously, the presign version called
+    /// [`PresignParticipant::initialize_presign_message()`], which does
+    /// some additional record-keeping of the associated auxinfo and keygen
+    /// SIDs. Does that still need to be done?
```

Comment:
note: no, not once we're single-sid. that map is used for maintaining a record of which keygen + auxinfo material to use for presign.